docs/docs/classic-ml/deep-learning/tensorflow/guide/index.mdx

```diff
@@ -0,0 +1,676 @@
+    )
+
+    model.compile(
+        optimizer=keras.optimizers.Adam(learning_rate=learning_rate),
+        loss="sparse_categorical_crossentropy",
+        metrics=["accuracy"],
+    )
+
+    return model
+
+
+def objective(trial):
+    # Start nested run for this trial
+    with mlflow.start_run(nested=True):
+        # Log hyperparameters
+        params = {
+            "learning_rate": trial.params["learning_rate"],
+            "units": trial.params["units"],
+            "dropout": trial.params["dropout"],
+        }
+        mlflow.log_params(params)
+
+        # Create and train model
+        model = create_model(trial)
+        history = model.fit(
+            x_train, y_train, validation_data=(x_val, y_val), epochs=5, verbose=0
+        )
+
+        # Get validation accuracy
+        val_accuracy = max(history.history["val_accuracy"])
+        mlflow.log_metric("val_accuracy", val_accuracy)
+
+        # Log model
+        mlflow.tensorflow.log_model(model, "model")
+
+        return val_accuracy
+
+
+# Main experiment run
+with mlflow.start_run(run_name="hyperparameter_optimization"):
+    # Log study parameters
+    mlflow.log_params(
+        {
+            "optimization_framework": "optuna",
+            "n_trials": 20,
+            "direction": "maximize",
+            "metric": "val_accuracy",
+        }
+    )
+
+    # Create and run study
+    study = optuna.create_study(direction="maximize")
+    study.optimize(objective, n_trials=20)
+
+    # Log best parameters and score
+    mlflow.log_params({f"best_{k}": v for k, v in study.best_params.items()})
+    mlflow.log_metric("best_val_accuracy", study.best_value)
+
+    # Train final model with best parameters
+    final_model = create_model(study.best_trial)
+    final_model.fit(x_train, y_train, epochs=10)
+    mlflow.tensorflow.log_model(final_model, "best_model")
+```
+
+### Deployment Preparation
+
+Prepare your TensorFlow model for deployment with MLflow:
+
+<details>
+  <summary>Model Serving with MLflow</summary>
+
+  ```python
+  import mlflow
+  import tensorflow as tf
+  from tensorflow import keras
+
+  # Create and train model
+  model = keras.Sequential([...])
+  model.compile(...)
+  model.fit(...)
+
+  # Log model with sample input for signature
+  with mlflow.start_run() as run:
+      # Create example input
+      sample_input = tf.random.uniform([1, 28, 28, 3])
+
+      # Add prediction signature
+      signature = mlflow.models.infer_signature(
+          sample_input.numpy(), model.predict(sample_input)
+      )
+
+      # Log model with sample input and signature
+      mlflow.tensorflow.log_model(
+          model,
+          "model",
+          signature=signature,
+          input_example=sample_input.numpy(),
+          pip_requirements=["tensorflow==2.10.0"],
+          code_paths=["./preprocessing.py"],  # Include custom code
+      )
+
+      model_uri = f"runs:/{run.info.run_id}/model"
+
+  # Deploy model to MLflow server
+  import subprocess
```

Comment:
Could we remove above model logging steps since they've been explained earlier? And simply put a bash command with `mlflow models serve`? Cuz i think that's easier for users who don't want to use subprocess and it's more like a general instruction

Reply:
Great feedback. Converting!